src/features/session_replay/instrument/index.js

```diff
@@ -78,12 +80,14 @@ export class Instrument extends InstrumentBase {
 
       // If startReplay() has been used by this point, we must record in full mode regardless of session preload:
       // Note: recorder starts here with w/e the mode is at this time, but this may be changed later (see #apiStartOrRestartReplay else-case)
-      this.recorder ??= new Recorder({ ...this, mode: this.#mode, agentRef: this.#agentRef, timeKeeper: this.#agentRef.runtime.timeKeeper }) // if TK exists due to deferred state, pass it
+      this.recorder ??= new Recorder({ ...this, mode: this.#mode, agentRef: this.#agentRef, trigger, timeKeeper: this.#agentRef.runtime.timeKeeper }) // if TK exists due to deferred state, pass it
       this.recorder.startRecording()
       this.abortHandler = this.recorder.stopRecording
     } catch (err) {
       this.parent.ee.emit('internal-error', [err])
     }
+    /** cancel the original import before preloading the agg if it's already been staged */
+    this.#importAbortController?.abort()
```

Comment:
I think this is all wrong. If the import has already happen, i.e. replay was already allowed to run or running, then it should just keep running. Rather, the API call should not reset and start over or do something. startReplay should only restart replays that are paused or stopped.

Reply:
Ah so then in that case, the API can start the recorder, but still not reset the aggregator import. The wiring for Aggregate args should be modified instead if need be.

Reply:
yep exactly.  just stage the agg import once like the other features do, but handle "early" importing the recorder for preload/api